precompile/allow_list.go

```diff
@@ -147,49 +146,55 @@ func (mal *modifyAllowListPrecompile) Run(evm PrecompileAccessibleState, callerA
 
 	// Verify that the caller is in the allow list and therefore has the right to modify it
 	callerStatus := GetAllowListStatus(evm.GetStateDB(), callerAddr)
-	if !callerStatus.IsAdmin() {
-		log.Info("EVM received attempt to modify the allow list from a non-allowed address", "callerAddr", callerAddr)
+	if !IsAllowListAdmin(callerStatus) {
+		log.Info("allow list EVM received attempt to modify the allow list from a non-allowed address", "callerAddr", callerAddr, "callerStatus", callerStatus)
 		return nil, remainingGas, fmt.Errorf("caller %s cannot modify allow list", callerAddr)
 	}
 
-	// Unpack the precompile's input into the arguments for setAllowListStatus
-	address, status, err := UnpackModifyAllowList(input)
-	if err != nil {
-		log.Info("modify allow list reverted", "err", err)
-		return nil, remainingGas, fmt.Errorf("failed to unpack modify allow list input: %w", err)
-	}
-
-	setAllowListStatus(evm.GetStateDB(), address, status)
+	SetAllowListRole(evm.GetStateDB(), modifyAddress, role)
 
 	// Return an empty output and the remaining gas
 	return []byte{}, remainingGas, nil
 }
 
-// RequiredGas returns the amount of gas consumed by this precompile.
-func (mal *modifyAllowListPrecompile) RequiredGas(input []byte) uint64 { return ModifyAllowListGasCost }
+// createAllowListSetter returns an execution function for setting the allow list status of the input address argument to [role].
```

Comment:
nit: add type of the execution function and explain how it is intended to be used to implement the given function signatures.